addons/crm/static/src/views/forecast_search_model.js

```diff
@@ -28,34 +28,22 @@ export class ForecastSearchModel extends SearchModel {
     /**
      * @override
      */
-    _getDomain(params = {}) {
-        const domain = super._getDomain(...arguments);
-        const forecastField = this.globalContext.forecast_field;
-        if (!forecastField) {
-            return domain;
+    _getSearchItemDomain(activeItem) {
+        let domain = super._getSearchItemDomain(activeItem);
+        const { searchItemId } = activeItem;
+        const searchItem = this.searchItems[searchItemId];
+        const context = makeContext([searchItem.context || {}]);
+        if (context.forecast_filter) {
+            const forecastField = this.globalContext.forecast_field;
```

Comment:
`forecast_filter` cannot work if there is no `forecast_field` So I don't know if it's a good idea to protect against a missing `forecast_field` (I would be happier as a dev with an error instead of nothing if I forgot to put it)
What do you think ?